gapic-common/lib/gapic/paged_enumerable.rb

```diff
@@ -107,6 +109,25 @@ def each_page
       end
     end
 
+    ##
+    # Iterate over the pages with the operation used to fetch the page.
+    #
+    # @yield [Page] Gives the pages in the stream.
+    #
+    # @raise [GapicError] if it's not started yet.
+    #
+    def each_page_with_operation
```

Comment:
Since `#operation` is already on `Page`, I don't think this offers much utility to the user.

My thought was to have a resource-centered method that also included operation. Sort of like [`Enumerable#each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). But, the more I have thought about this, the more I wonder how it should behave. Does it return the same `operation` for each resource in the page? Or does it return the `operation` for only the first resource on the page?  What if the page has an empty list of resources but has `operation`? (Is that even possible?)

I'd be fine removing `each_page_with_operation`. I think I would still favor a `each_with_operation` method, but we should probably discuss the behavior in more detail.